src/arithmetic/fields.rs

```diff
@@ -1,10 +1,17 @@
 //! This module contains the `Field` abstraction that allows us to write
 //! code that generalizes over a pair of fields.
 
+use core::mem::size_of;
+use static_assertions::const_assert;
+use std::assert;
+use std::convert::TryInto;
+use std::marker::PhantomData;
 use subtle::{Choice, ConstantTimeEq, CtOption};
 
 use super::Group;
 
+const_assert!(size_of::<usize>() >= 4);
```

Comment:
Let's remove

```rust
// No goofy hardware please.
assert!(core::mem::size_of::<usize>() >= 4);
```

from `commitment.rs` since we're making this static now.